### PR TITLE
feat: add vsa e2e

### PR DESCRIPTION
**Pull request by ramonpetgrave64**
followup to https://github.com/slsa-framework/slsa-verifier/pull/777

adds e2e tests for VSAs

## Testing

- I temporarily modified this workflow to also trigger and auto-open and close issues in PRs.
- edited the script to be able to test failure cases
- after creating the a new issue laebl "area:vsa" in slsa-framework/slsa-verifier, the workflow did auto-open and aut-close an issue.
  - https://github.com/slsa-framework/slsa-verifier/issues/792

example successful run
 - https://github.com/slsa-framework/example-package/actions/runs/10062146398/job/27813925490#step:6:9

example failed run
 - https://github.com/slsa-framework/example-package/actions/runs/10062068638/job/27813655378#step:6:9

### CODE REVIEW

**Comment by ramonpetgrave64** [comment]
@haydentherapper

File: .github/workflows/scripts/e2e.vsa.gcp.gke.verify.sh [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        --public-key-id keystore://76574:prod:vsa_signing_public_key \
+        --public-key-path ./vsa_signing_public_key
+}
+
+iterate() {
+    # find all attestations
+    declare -a FILE_PATHS
+    while IFS= read -r file; do
+        FILE_PATHS+=("$file")
+    done < <(find ./gke-node-images:238739202978 ./gke-master-images:78064567238 -type f)
```
Comment (haydentherapper) [inline .github/workflows/scripts/e2e.vsa.gcp.gke.verify.sh:27]: Would it be better to search for all intoto.jsonl files, so if they add other top level directories in the future, this script will pick them up?

**Review by Hayden-IO — COMMENTED** [review]
Just wondering, about how long does this take to run, given it's verifying ~3000 VSAs?

File: .github/workflows/scripts/e2e.vsa.gcp.gke.verify.sh (hunk 1 above)
Comment (ramonpetgrave64) [reply .github/workflows/scripts/e2e.vsa.gcp.gke.verify.sh:27]: By the time they add support for container images, I suspect they may change the structure of the directories, so I hesitate to simple look for all  intoto.jsonl files.

**Comment by ramonpetgrave64** [comment]
> Just wondering, about how long does this take to run, given it's verifying ~3000 VSAs?

About 38s to run slsa-verifier against all the VSAs, and about 2m for the whole workflow.

File: .github/workflows/scripts/e2e.vsa.gcp.gke.verify.sh (hunk 1 above)
Comment (haydentherapper) [reply .github/workflows/scripts/e2e.vsa.gcp.gke.verify.sh:27]: Looks like we fail if we find no files, so SG, this should be sufficient and will catch if they make changes in the future.

**Review by Hayden-IO — APPROVED** [review]

**Pull request merged**